services/explorer/src/pages/TransactionData/EventDataDisplay.tsx

```diff
@@ -0,0 +1,55 @@
+import { BaseToggleButtonGroup, Box } from '@taraxa_project/taraxa-ui';
+import { DataRow } from '../../components';
+import { useState } from 'react';
+
+export interface EventDataProps {
+  hexValue: string;
+}
+
+export const EventDataDisplay = ({ hexValue }: EventDataProps) => {
+  const toggleValues = [
+    { value: 'hex', label: 'Hex' },
+    { value: 'dec', label: 'Dec' },
+  ];
+
+  const [decoded, setDecoded] = useState<string>(hexValue);
+  const [selected, setSelected] = useState<string>(toggleValues[0].value);
+
+  const onChange = (
+    event: React.MouseEvent<HTMLElement>,
+    newValue: string | null
+  ) => {
+    const found = toggleValues.find((toggle) => toggle.value === newValue);
+
+    setSelected(found.value);
+    switch (newValue) {
+      case toggleValues[0].value:
+        setDecoded(hexValue);
+        break;
+      case toggleValues[1].value:
+        const addressHex = '0x' + hexValue.slice(-40);
+        const addressDecimal = BigInt(addressHex).toString();
+        setDecoded(addressDecimal);

```

Comment:
![image](https://github.com/Taraxa-project/taraxa-platform/assets/100562086/412ce3b4-297e-417c-8ba4-e98310be9032)

This fails on empty data

Reply:
Also, this won't work like this as we don't know the primitive that comes in here as base, so we should remove this. The main idea for this was to add a `decimal <> hex switch to the tx values and the block numbers` for debug purposes.